app/src/main/java/com/example/triptracker/view/map/RecordScreen.kt

```diff
@@ -245,9 +254,39 @@ fun Map(
               onClick = {
                 if (viewModel.isRecording()) {
                   viewModel.stopRecording()
+
+                  // Add itinerary to database
+                  val id = itineraryRepository.getUID()
+                  val title = "TEST" // TODO : get title from user but not implemented yet
+                  val username = "lomimi" // TODO : get username from user but not implemented yet
+                  val location =
+                      Location(deviceLocation.latitude, deviceLocation.longitude, "Device Location")
+                  // TODO : get location from user but not implemented yet (default device location)
+                  val flameCount = 0L
+                  val startDate = viewModel.startDate.value
+                  val endDate = viewModel.endDate.value
+                  val pinList =
+                      emptyList<Pin>() // TODO : get pin list from user but not implemented yet
+                  val description =
+                      "description" // TODO : get description from user but not implemented yet
+
+                  val itinerary =
```

Comment:
This looks good, I'm just wondering if this shouldn't be added in some sort of coroutine to do it asynchronously. Through the view model for instance